core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1593,6 +1594,22 @@ public void addView(View v) throws IOException {
         viewGroupMixIn.addView(v);
     }
 
+    /**
+     * Completely replaces views
+     */
+    @DataBoundSetter
+    public void setViews(Collection<View> views) throws IOException {
```

Comment:
then :bug: as it requires at least some syncronization as at the moment you can get a junk result.